### PR TITLE
Scribbles - Copy WriteLogits functionality to generic segmentation app for non-coldstart scribbles methods

### PR DESCRIPTION
**Pull request by masadcv**
With the recent moving of scribbles functionality into generic segmentation app, I seem to have missed the WriteLogits functionality needed for non-coldstart scribbles options. 

This PR attempts to fix this.

- [x] Adding WriteLogits transform to Inferer
- [x] Adding additional code to main.py for handling datastore queries for saving/loading logits
- [x] Tested all three scribbles functionalities within the generic segmentation apps, this includes  1) coldstart, 2) graphcut + logits and 3) crf + logits 

Signed-off-by: masadcv <muhammad.asad@kcl.ac.uk>

### CODE REVIEW

File: sample-apps/segmentation/main.py [hunk 1]
```diff
@@ -80,3 +88,27 @@ def init_strategies(self):
             "random": Random(),
             "first": MyStrategy(),
         }
+
```
Comment (SachidanandAlle) [inline sample-apps/segmentation/main.py:91]: Lets discuss... probably we want to avoid this for public release.. and keep all generic scribbles for all apps that doesn't have app specific dependency (in this case infer output from other infer tasks)

Comment (masadcv) [reply sample-apps/segmentation/main.py:91]: Sure, lets discuss in our meeting. This infer function works just fine for non scribbles tasks. The dependency is broken when we try to run scribbles method that need logits from CNN. Maybe there is a more elegant way of achieving this....

Comment (masadcv) [reply sample-apps/segmentation/main.py:91]: Another thought, maybe we can only keep coldstart for this generic app - which wont require logits. Then perhaps we may want to keep atleast one app around that covers/showcases all scribbles functionality (e.g. **segmentation_spleen_scribbles** that is due to be removed by end of this week)

**Comment by masadcv** [comment]
As discussed in the weekly meeting, we are now going to remove all model-based scribbles methods and only keep model-free (coldstart) option for release 0.2. Therefore I am going to remove this and other methods for now and revisit after the release.
